src/wallet/IndyWallet.ts

```diff
@@ -22,86 +22,267 @@ import { AriesFrameworkError } from '../error'
 import { JsonEncoder } from '../utils/JsonEncoder'
 import { isIndyError } from '../utils/indyError'
 
+import { WalletDuplicateError, WalletNotFoundError, WalletError } from './error'
+
+export interface IndyOpenWallet {
+  walletHandle: number
+  masterSecretId: string
+  walletConfig: WalletConfig
+  walletCredentials: WalletCredentials
+}
```

Comment:
I'm note sure if this type has right cohesion. It seems these attributes are kind of unrealated, at least from the first point of view 🤷‍♂️

Reply:
It contains related to an open wallet. I agree with your point, but would like to solve that in a separate PR. 

I think we can make the wallet a bit more straightforward by splitsing the functionality in two parts:

- A "controller" or something that does the following
	- create
	- open
	- delete
	- close
- A wallet that does the actual wallet stuff. You need an open wallet to use it.

This will make it easier to handle multiple wallets I think and separates the lifecycle of a wallet from the usage of the wallet

I'd like to do that in a future PR

Reply:
Cool, good idea 👍